### PR TITLE
Shrink our gigantic tests!

### PR DESCRIPTION
**Pull request by Wafflespeanut**
This makes use of a simple wrapper around the JSON object and marks stuff whenever we 'get' something from the dictionary (using `[]`). This can help shrink the gigantic tests we currently have (which has a hell lot of unused key/val)

### CODE REVIEW

**Comment by Wafflespeanut** [comment]
gotta fix the flake8 stuff...

**Comment by Wafflespeanut** [comment]
It might be worth taking a look at the failed Travis build! :smile:

**Comment by Wafflespeanut** [comment]
@jdm I think it's still buggy, but well, this is just a prototype. I'm curious whether this would be useful if I make it to work.

**Comment by Wafflespeanut** [comment]
Now, [this](https://travis-ci.org/servo/highfive/builds/128450660) is progress! (still, 3 tests fail)

**Comment by Wafflespeanut** [comment]
All the tests pass now, but I had to put some type conversions to integrate the lint.

**Comment by Wafflespeanut** [comment]
After trying to integrate this into highfive (for a while now), I realize that the entire problem lies on creating the tests, where we read the JSON files twice (one for the initial/expected stuff), and the other for payload. I think I should modify the linter to take objects instead of going directly for the file.

**Comment by Wafflespeanut** [comment]
- fixed the linter for flagging only the stuff in `payload`
- avoided reading the test JSONs twice
- added flag to warn/overwrite the nodes in test files

only the command-line argument is left to go...

**Comment by Wafflespeanut** [comment]
I'm finally satisfied with [the output](https://travis-ci.org/servo/highfive/builds/128686632) :smile:

**Comment by Wafflespeanut** [comment]
This is ready for review!

File: handlers/assign_reviewer/__init__.py [hunk 1]
```diff
@@ -55,7 +55,7 @@ def on_pr_opened(self, api, payload):
         pr = payload["pull_request"]
         # If the pull request already has an assignee,
         # don't try to set one ourselves.
-        if pr["assignee"] != None:
+        if pr["assignee"] != None:      # NOQA (silence flake8 here)
```
Comment (jdm) [inline handlers/assign_reviewer/__init__.py]: What is it complaining about?

Comment (Wafflespeanut) [inline handlers/assign_reviewer/__init__.py]: flake8 prefers `is not None`, but we need `!=` to call our overridden `__ne__` method (the former checks for an exact match of the object, which will fail while testing)

**Comment by jdm** [comment]
The need for `str(foo)` and `int(foo)` in various places is disappoint. Is there no way around that?

**Comment by Wafflespeanut** [comment]
True, they *are* bad, but I couldn't find a way around them. I've fixed some of them, and now we have three situations (in regex matching) where `str(foo)` occurs. Python's `re` methods look pretty strict, that they accept only string/unicode, which prevents us from overriding anything :disappointed:

**Comment by jdm** [comment]
Thanks!

**Comment by Wafflespeanut** [comment]
Thanks for reviewing it :)

**Pull request merged**